Standardize parameters for GUNW

cmarshak opened the pull request:
* Require `frame_id`
* Set default parameters for version 3+ GUNW products
* Expose additional customizable parameters and specify "standard" parameters where possible.

cmarshak commented:
@jhkennedy and @asjohnston-asf, I know we should not merge this until https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp/pull/122 is into dev.

However, wanted to get your input now. I modified some pieces including requiring frame-id - think this will ensure operators don't screw up a big processing campaign going forward - again making frames required for "standard" products.

@dbekaert - tagging you just so you are aware.

jhkennedy commented:
@cmarshak it looks like the released version of DockerizedTopsApp accepts a `--frame-id` parameter, but the `--output-resolution`, `--dense-offsets`, and `--goldstein-filter-power` parameters haven't been released yet. 

Should we go ahead and update `INSAR_ISCE` with these changes, where they can be? (fame id, iono delay defaults/description, esd description, sed default)

cmarshak commented on job_spec/INSAR_ISCE_TEST.yml:
```diff
@@ -27,31 +28,50 @@ INSAR_ISCE_TEST:
+      api_schema:
+        description: Subset GUNW products to this frame. Standard products require frame id to be specified. When set to `-1`, no subsetting applied. If no frame specified (i.e. `-1`), then output GUNW is not a standard GUNW. 
+        default: '""'
+        type: integer
```
Thanks, Joe!

cmarshak commented:
> @cmarshak it looks like the released version of DockerizedTopsApp accepts a `--frame-id` parameter, but the `--output-resolution`, `--dense-offsets`, and `--goldstein-filter-power` parameters haven't been released yet.
> 
> Should we go ahead and update `INSAR_ISCE` with these changes, where they can be? (fame id, iono delay defaults/description, esd description, set default)

Whatever is easier. 

It won't matter too much from my vantage point because all these "new paramters" will never generate products that are delivered to DAAC.

jhkennedy commented:
> Whatever is easier.
>
> It won't matter too much from my vantage point because all these "new paramters" will never generate products that are delivered to DAAC.

@cmarshak I believe the reasoning to require including frame_id or not applies even more so to the *production* job type, not the test type, no?

Similarly, the `*_TEST` job should be for *testing*, not operational or user production. So, if you need the custom parameters, you'll want to expose them in either the production job type, or a custom-production job type, leaving the test job type for development so you can evaluate how developmental changes affect the products.

cmarshak commented:
> > Whatever is easier.
> > It won't matter too much from my vantage point because all these "new paramters" will never generate products that are delivered to DAAC.
> 
> @cmarshak I believe the reasoning to require including frame_id or not applies even more so to the _production_ job type, not the test type, no?
> 
> Similarly, the `*_TEST` job should be for _testing_, not operational or user production. So, if you need the custom parameters, you'll want to expose them in either the production job type, or a custom-production job type, leaving the test job type for development so you can evaluate how developmental changes affect the products.

Sorry for my miscommunication - yes `frame_id` needs to be in production as that will be how differentiate the standard product!  I think we are exposing more parameters that will eventually be wrapped into other standardized products.

Keep it easy and put them in both, I'd say.

jhkennedy commented on job_spec/INSAR_ISCE_TEST.yml:
```diff
@@ -87,6 +107,14 @@ INSAR_ISCE_TEST:
         - Ref::compute_solid_earth_tide
+        - --output-resolution
+        - Ref::output_resolution
+        - --dense-offsets
```
@cmarshak I don't see an API parameter for `dense_offsets`

cmarshak commented:
Hey Joe and ASF team,

I fixed the dense offset issues, made TEST and PROD job specs the same, and changed "standard product" to "standard ARIA product".

I think making the job specs the same will make it easier. Hopefully adding a swagger plugin will prevent similar oversights on my minimal contributions. Thank you again for your patience and help with this!

jhkennedy reviewed (approved):
@cmarshak I think this is ready now, so I'll merge and run some tests. 

Notably, we can't/shouldn't list API specs for options that aren't used/available yes, so I've removed the API spec for  `output_resolution`, `dense_offsets`, `goldstein_filter_power` and `unfiltered_coherence` from `INSAR_ISCE.yml`, but otherwise `INSAR_ISCE` and `INSAR_ISCE_TEST` match. 

Once DockerizedTopsApp is released with the new option we can add them back.
